code/game/machinery/weathercontrol.dm

```diff
@@ -0,0 +1,183 @@
+		/obj/item/weapon/cell/rad,
+		/obj/item/weapon/stock_parts/matter_bin/adv/super/bluespace,
+		/obj/item/weapon/stock_parts/micro_laser/high/ultra,
+		/obj/item/weapon/stock_parts/micro_laser/high/ultra)
+	RefreshParts()
+
+/obj/machinery/weathercontrol/Destroy()
+	if(crystal_reserve >= 300)
+		new /obj/item/bluespace_crystal/flawless(loc)
+	else
+		while(crystal_reserve >= 3)
+			new /obj/item/bluespace_crystal(loc)
+			crystal_reserve -= 3
+		while(crystal_reserve > 0)
+			new /obj/item/bluespace_crystal/artificial(loc)
+			crystal_reserve--
+	..()
+
+/obj/machinery/weathercontrol/RefreshParts()
+	var/T = 0
+	for(var/obj/item/weapon/stock_parts/micro_laser/ML in component_parts)
+		T += ML.rating
+	efficiency_modifier = (T/2)-3
+	//Both tier 3: 0; one tier 4: 0.5, both tier 4: 1
+
+/obj/machinery/weathercontrol/update_icon()
+	return
+
+/obj/machinery/weathercontrol/attack_hand(mob/user)
+	return ui_interact(user)
+
+/obj/machinery/weathercontrol/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open=NANOUI_FOCUS)
+	if (gcDestroyed || !get_turf(src) || !anchored)
+		if(!ui)
+			ui = nanomanager.get_open_ui(user, src, ui_key)
+		if(ui)
+			ui.close()
+		return
+
+	var/data[0]
+	data["error"] = error_message
+
+	if(map.climate)
+		var/datum/climate/C = map.climate
+		var/datum/weather/CW = C.current_weather
+		var/forecast_dat
+		for(var/datum/weather/W in C.forecasts)
+			forecast_dat += "[W.name] "
+		data["name"] = name
+		data["currentweather"] = CW
+		data["remaining_time"] = formatTimeDuration(CW.timeleft)
+		data["forecast"] = forecast_dat
+		data["crystals"] = crystal_reserve
+	else
+		data["error"] = "Error: No climate detected."
+
+	// update the ui if it exists, returns null if no ui is passed/found
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "weathercontrol.tmpl", "Weather Control Device", 380, 250)
+		ui.set_initial_data(data)
+		ui.open()
+
+#define DISRUPT_COST 1
+#define INTENSIFY_COST 1
+#define ABATE_COST 2
+
+#define NOFIRE 2
+#define NEED_CRYSTALS -3
+#define POWER_ERROR -4
+/obj/machinery/weathercontrol/Topic(href, href_list)
+	if(..())
+		return
+	if(usr.incapacitated() || (!Adjacent(usr)&&!isAdminGhost(usr)) || !usr.dexterity_check() || !map.climate)
+		return
+	if(!allowed(usr) && !emagged)
+		to_chat(usr,"<span class='warning'>Access denied.</span>")
+		return
+	if(stat & NOPOWER)
+		return
+	var/datum/climate/C = map.climate
+	var/datum/weather/CW = C.current_weather
+
+	var/feedback = NOFIRE
+	var/usedcost = 0
+
+	if(stat & NOPOWER)
+		feedback = POWER_ERROR
+	else
+		if(href_list["disrupt"])
+			feedback = TRUE
+			usedcost = DISRUPT_COST
+			if(!burn_crystals(DISRUPT_COST))
+				feedback = NEED_CRYSTALS
+			else
+				if(C.current_weather.next_weather.len < 2)
+					feedback = CANNOT_CHANGE
+				else
+					CW.timeleft = min(1 MINUTES, CW.timeleft)
+					C.forecast()
+		if(href_list["intensify"])
+			feedback = TRUE
```

Comment:
I don't like using this define for an obviously non-boolean variable

Reply:
It just means success

Reply:
Giving me R0B0's `TRUE + 1` vibes